Failed tests (12478): TestStoreRangeRemoveDead

The following test appears to have failed:

[#12478](https://circleci.com/gh/cockroachdb/cockroach/12478):

```
1      kv/txn_coord_sender.go:756
[Trace t923d1aa3-9e78-4485-9e56-347071e871e6]
W0208 14:51:45.628551 708 rpc/server.go:345  rpc: write response failed: write tcp 127.0.0.1:43575->127.0.0.1:35094: use of closed network connection
I0208 14:51:45.628714 708 storage/replica_command.go:463  auto-gc'ed 923d1aa3 (3 intents)
I0208 14:51:45.629108 708 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
panic: assignment to entry in nil map

goroutine 8803 [running]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).addInternal(0xc82021cb00, 0xc8200dc0d0, 0xc820c00201, 0x3ff0000000000000, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:312 +0x7d5
github.com/cockroachdb/cockroach/storage.(*baseQueue).Add(0xc82021cb00, 0xc8200dc0d0, 0x3ff0000000000000, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:245 +0xbd
github.com/cockroachdb/cockroach/storage.(*Replica).changeReplicasTrigger.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/replica_command.go:1644 +0x5a
github.com/cockroachdb/cockroach/storage/engine.(*rocksDBBatch).Commit(0xc820c7ef60, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:471 +0x194
github.com/cockroachdb/cockroach/storage.(*Replica).applyRaftCommand(0xc8200dc0d0, 0x7f468c730d40, 0xc820c7ef30, 0x12, 0x100000001, 0x1, 0x989680, 0x5, 0x100000001, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1293 +0x677
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8202d10a0, 0xc820c005f0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 1 [chan receive]:
testing.RunTests(0x18e9610, 0x1fc6480, 0xe3, 0xe3, 0x1)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc820995ef8, 0x1)
	/usr/local/go/src/testing/testing.go:494 +0x70
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820995ef8)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x21
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820995ef8)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:38 +0x26
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:514 +0x113

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x230ec40)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0xbc

goroutine 8740 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc820ce21e0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 8916 [runnable]:
syscall.Syscall(0x1, 0x2, 0xc820315c00, 0x5c, 0x5c, 0x5c, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.write(0x2, 0xc820315c00, 0x5c, 0x1c0, 0x230e4c0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:1064 +0x5f
syscall.Write(0x2, 0xc820315c00, 0x5c, 0x1c0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:176 +0x4d
os.(*File).write(0xc820028018, 0xc820315c00, 0x5c, 0x1c0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:232 +0xaa
os.(*File).Write(0xc820028018, 0xc820315c00, 0x5c, 0x1c0, 0x1c0, 0x0, 0x0)
	/usr/local/go/src/os/file.go:139 +0x8a
--
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop.func1(0xc82000c280, 0xc8204e3c80)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:275 +0x3d2
created by github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:279 +0x6f

goroutine 8752 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8203c8550)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 8776 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc820ce2500)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 8764 [select]:
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop(0xc82000c280)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:281 +0x2bc
github.com/cockroachdb/cockroach/storage_test.TestStoreRangeRemoveDead(0xc820246000)
	/go/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:1169 +0xaba
testing.tRunner(0xc820246000, 0x1fc76f8)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d
FAIL	github.com/cockroachdb/cockroach/storage	11.089s
=== RUN   TestBatchBasics
I0208 14:51:22.469394 667 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I0208 14:51:22.470576 667 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I0208 14:51:22.473202 667 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -1165,7 +1165,6 @@ func TestStoreRangeRemoveDead(t *testing.T) {
 			mtc.stores[1].ForceReplicationScanAndProcess()
 		}
 	}
-	ticker.Stop()
 }
 
 // TestStoreRangeRebalance verifies that the replication queue will take

--- a/storage/queue.go
+++ b/storage/queue.go
@@ -382,11 +382,6 @@ func (bq *baseQueue) processLoop(clock *hlc.Clock, stopper *stop.Stopper) {
 
 			// Exit on stopper.
 			case <-stopper.ShouldStop():
-				bq.mu.Lock()
-				bq.mu.replicas = nil
-				bq.mu.priorityQ = nil
-				bq.mu.purgatory = nil
-				bq.mu.Unlock()
 				return
 			}
 		}

